homeassistant/components/sensor/phicomm.py

```diff
@@ -0,0 +1,228 @@
+        self._hass = hass
+        self._username = username
+        self._password = password
+        self._token_path = hass.config.path(TOKEN_FILE + username)
+        self._token = None
+        self._devices = None
+        self.devs = None
+
+    @asyncio.coroutine
+    def make_sensors(self, name, sensors):
+        """Make sensors with online data."""
+        try:
+            with open(self._token_path) as file:
+                self._token = file.read()
+                _LOGGER.debug("Load: %s => %s", self._token_path, self._token)
+        except BaseException:
+            pass
+
+        self.update_data()
+        if not self.devs:
+            return None
+
+        devices = []
+        for index in range(len(self.devs)):
+            for sensor_type in sensors:
+                devices.append(PhicommSensor(self, name, index, sensor_type))
+        self._devices = devices
+        return devices
+
+    @asyncio.coroutine
+    def async_update(self, time):
+        """Update online data and update ha state."""
+        old_devs = self.devs
+        self.update_data()
+
+        tasks = []
+        for device in self._devices:
+            if device.state != device.state_from_devs(old_devs):
+                _LOGGER.info('%s: => %s', device.name, device.state)
+                tasks.append(device.async_update_ha_state())
+
+        if tasks:
+            yield from asyncio.wait(tasks, loop=self._hass.loop)
+
+    def update_data(self):
+        """Update online data."""
+        try:
+            json = self.fetch_data()
+            if ('error' in json) and (json['error'] != '0'):
+                _LOGGER.debug("Reset token: error=%s", json['error'])
+                self._token = None
+                json = self.fetch_data()
+            self.devs = json['data']['devs']
+            _LOGGER.info("Get data: devs=%s", self.devs)
+        except BaseException:
+            import traceback
+            _LOGGER.error('Exception: %s', traceback.format_exc())
+
+    def fetch_data(self):
+        """Fetch the latest data from Phicomm server."""
+        if self._token is None:
+            import hashlib
+            md5 = hashlib.md5()
+            md5.update(self._password.encode('utf8'))
+            data = {
+                'authorizationcode': AUTH_CODE,
+                'phonenumber': self._username,
+                'password': md5.hexdigest().upper()
+            }
+            headers = {'User-Agent': USER_AGENT}
+            json = requests.post(TOKEN_URL, headers=headers, data=data).json()
+            _LOGGER.debug("Get token: %s", json)
+            if 'access_token' in json:
+                self._token = json['access_token']
+                with open(self._token_path, 'w') as file:
+                    file.write(self._token)
+            else:
+                return None
+        headers = {'User-Agent': USER_AGENT, 'Authorization': self._token}
+        return requests.get(DATA_URL, headers=headers).json()
```

Comment:
requests isn't async, you need to look into aiohttp, you should also use hass' aio session and not create your own